File: app/src/main/java/com/waygo/data/model/butler/ButlerSayResponse.java
package com.waygo.data.model.butler;


import android.support.annotation.NonNull;

import com.waygo.utils.result.ResultJ;
import com.waygo.utilskt.Result;

public final class ButlerSayResponse extends ButlerResponse {

    private ButlerSayResponse(@NonNull final String message) {
        super(message);
    }

    @NonNull
    public static Result<ButlerResponse> create(@NonNull final String message) {
        return createMessage(message)
                .map(s -> (ButlerResponse) new ButlerSayResponse(s));
    }

}

File: app/src/main/java/com/waygo/data/model/butler/ButlerSayResponse.kt
package com.waygo.data.model.butler


import com.waygo.utilskt.Result
import kotlin.platform.platformStatic

public class ButlerSayResponse private constructor(message: String) : ButlerResponse(message) {
    companion object {

        public platformStatic fun create(message: String): Result<ButlerSayResponse> =
                ButlerResponse.createMessage(message)
                        .map { ButlerSayResponse(it) }
    }

}
